### PR TITLE
Instances List improvements

### PR DESCRIPTION
**Pull request by rs4231199**
for #3219

- [x] Add the ability to sort by clicking on the names of the columns (clicking twice reverses the sort)
- [ ] ~Add the ability to select an instance and hit the delete key (and confirm delete)~
- [x] Add the ability to lock/unlock by clicking on the lock icon
- [x] Add the ability to multi-select (great for mass deletion)

### CODE REVIEW

**Comment by arthuro555** [comment]
GitHub automatically requests a review when opening a non-draft PR because of the code owners file

**Comment by rs4231199** [comment]
> Thanks for the PR! You've asked for the review already, do you want me to check this now or later when adding the other improvements?
@4ian 
I think there is no common code between this functionality and the rest. So, a review now is possible.
Or you can do it later. It doesn't really matter. I hope to push more changes today. :)

**Comment by rs4231199** [comment]
> Add the ability to lock/unlock by clicking on the lock icon

Should I add unlock emoji(:unlock: ) so that users can click on it to lock the instance?

image

Img: This is what it looks like.

**Comment by 4ian** [comment]
@rs4231199 Good question, I would say consider:
- Either move to using a (small) `IconButton` with an icon from Material UI like LockOpen and Lock:
  image
  
- Or use a unlock emoji *but* use some opacity so that it's not hard to find which things are locked from those who are not.

But I think ideally a IconButton with `size="small"` would be good. Because an emoji is not supposed to be clickable, there won't be any support for hover, press, keyboard focus, i.e: it would not be accessible :)

File: newIDE/app/src/SceneEditor/index.js
Comment (4ian) [inline newIDE/app/src/SceneEditor/index.js:550]: Seems that this 3rd argument is unused (always true), we might think of removing it.

**Review by 4ian — COMMENTED** [review]
Looks good so far! :)

File: newIDE/app/src/SceneEditor/index.js
Comment (rs4231199) [reply newIDE/app/src/SceneEditor/index.js]: Yes, It wasn't used anywhere. I removed it.

**Comment by arthuro555** [comment]
> I was thinking of implementing this by maintaining some history that would allow undo and redo operations.

I started a discussion about the undo stack here: https://github.com/4ian/GDevelop/discussions/2739#discussioncomment-1794562

**Comment by 4ian** [comment]
@rs4231199 You're right that this is a problem because we're in the case of editors sharing the same "data" and so undo/redo is not consistent.
I would argue that we don't want to address this in this PR. Let's merge this even if we do not handle deletion (or if we do, without undo/redo for now). 

We can think about how to better handle undo/redo across editors separately (the discussion that Arthuro555 linked :))

**Comment by rs4231199** [comment]
I left a console.log statement in the code by mistake. Otherwise it is
ready for a review.

Do we want a ctrl+A(select all) shortcut? Users can first filter instances
using search bar situated below instances list. Then they can select all
filtered instances.
It would be useful if users want to lock/unlock all instances containing a
certain keyword(let's say "button").

File: newIDE/app/src/InstancesEditor/InstancesList/index.js
Comment (4ian) [inline newIDE/app/src/InstancesEditor/InstancesList/index.js:169]: Move this out of the function and out of the component. This will help reducing the pressure of the JS engine garbage collector by avoiding to re-declare a function.
Also use no abbreviations: `compareStrings`.

File: newIDE/app/src/InstancesEditor/InstancesList/index.js
Comment (4ian) [inline newIDE/app/src/InstancesEditor/InstancesList/index.js:134]: Use a ternary operator :)

File: newIDE/app/src/InstancesEditor/InstancesList/index.js
Comment (4ian) [inline newIDE/app/src/InstancesEditor/InstancesList/index.js:85]: Could you store the boolean directly instead?

File: newIDE/app/src/InstancesEditor/InstancesList/index.js
Comment (4ian) [inline newIDE/app/src/InstancesEditor/InstancesList/index.js:183]: See my other comment, prefer two booleans ;)

**Review by 4ian — COMMENTED** [review]
I've left a few comments. Once they are addressed and the console.log removed, I think it's good to go!

> Do we want a ctrl+A(select all) shortcut? Users can first filter instances
using search bar situated below instances list. Then they can select all
filtered instances.

Do you think it's easy to implement cleanly? If yes we could have it. Otherwise I would keep it for another PR!

**Comment by rs4231199** [comment]
> > Do we want a ctrl+A(select all) shortcut? Users can first filter instances
> > using search bar situated below instances list. Then they can select all
> > filtered instances.
> 
> Do you think it's easy to implement cleanly? If yes we could have it. Otherwise I would keep it for another PR!

Not implementing it, because `ctrl+A` is used to select the text from search bar.

PR should be good to merge now.

**Review by 4ian — APPROVED** [review]
LGTM! Thanks for working on this :)

**Pull request merged**
